examples/Example14/AdeptIntegration.cpp

```diff
@@ -0,0 +1,290 @@
+  else if (pdg == 22)
+    fBuffer.ngammas++;
+
+  if(fBuffer.toDevice.size() >= fBufferThreshold) 
+    { 
+      if (fDebugLevel > 0)
+        std::cout << "Reached the threshold of " << fBufferThreshold << " triggering the shower" << std::endl;
+      this->Shower(G4EventManager::GetEventManager()->GetConstCurrentEvent()->GetEventID()); 
+    }
+}
+
+void AdeptIntegration::Initialize(bool common_data)
+{
+  if (fInit) return;
+  if (fMaxBatch <= 0)
+    throw std::runtime_error("AdeptIntegration::Initialize: Maximum batch size not set.");
+
+  fNumVolumes = vecgeom::GeoManager::Instance().GetRegisteredVolumesCount();
+  // We set the number of sensitive volumes equal to the number of placed volumes. This is temporary
+  fNumSensitive = vecgeom::GeoManager::Instance().GetPlacedVolumesCount();
+  if (fNumVolumes == 0)
+    throw std::runtime_error("AdeptIntegration::Initialize: Number of geometry volumes is zero.");
+
+  if (common_data) {
+    std::cout << "=== AdeptIntegration: initializing geometry and physics\n";
+    // Initialize geometry on device
+    if (!vecgeom::GeoManager::Instance().IsClosed())
+      throw std::runtime_error("AdeptIntegration::Initialize: VecGeom geometry not closed.");
+
+    const vecgeom::cxx::VPlacedVolume *world = vecgeom::GeoManager::Instance().GetWorld();
+    if (!InitializeGeometry(world))
+      throw std::runtime_error("AdeptIntegration::Initialize: Cannot initialize geometry on GPU");
+
+    // Initialize G4HepEm
+    if (!InitializePhysics())
+      throw std::runtime_error("AdeptIntegration::Initialize cannot initialize physics on GPU");
+
+    // Do the material-cut couple index mapping once
+    // as well as set flags for sensitive volumes and region
+    int *sensitive_volumes = nullptr;
+
+    VolAuxData *auxData = CreateVolAuxData(G4TransportationManager::GetTransportationManager()->
+                                           GetNavigatorForTracking()->GetWorldVolume(),
+                                           vecgeom::GeoManager::Instance().GetWorld(), 
+                                           *fg4hepem_state); 
+
+    // Initialize volume auxiliary data on device
+    VolAuxArray::GetInstance().fNumVolumes = fNumVolumes;
+    VolAuxArray::GetInstance().fAuxData    = auxData;
+    VolAuxArray::GetInstance().InitializeOnGPU();
+    return;
+  }
+
+  std::cout << "=== AdeptIntegration: initializing transport engine for thread: " << G4Threading::G4GetThreadId()
+            << std::endl;
+
+  // Initialize user scoring data
+  fScoring = new AdeptScoring(fNumSensitive);
+  fScoring_dev = fScoring->InitializeOnGPU();
+
+  // Initialize the transport engine for the current thread
+  InitializeGPU();
+
+  fInit = true;
+}
+void AdeptIntegration::InitBVH()
+{
+  vecgeom::cxx::BVHManager::Init();
+  vecgeom::cxx::BVHManager::DeviceInit();
+}
+
+void AdeptIntegration::Cleanup()
+{
+  if (!fInit) return;
+  // if (!fCleanup.test_and_set()) {
+  //  std::cout << "=== Cleaning up GPU ...\n";
```

Comment:
Is this intentionally commented out and `fCleanup` not used at all?

Reply:
Indeed, not needed any more. Fixed.